decentraland/unity-renderer
# refactor: move SceneController and ECS interfaces to own assembly

## What this PR includes?

This PR is part of the following series to address https://github.com/decentraland/unity-renderer/issues/416:
* refactor: move SceneController and ECS interfaces to own assembly https://github.com/decentraland/unity-renderer/pull/429
* refactor: put all runtime context interfaces into own assembly https://github.com/decentraland/unity-renderer/pull/432
* refactor: put all platform context interfaces into own assembly https://github.com/decentraland/unity-renderer/pull/431
* refactor: put environment into own assembly https://github.com/decentraland/unity-renderer/pull/433

---

This PR manages to put all interfaces related to scene runtime (`IParcelScene`, `IDCLEntity`, `ISceneController`, `IComponent`, etc) into a single assembly: `WorldRuntimeInterfaces`. This is a step needed in order to move `Environment` to its own assembly.

- Logic changes are almost non-existent. This is pretty much code and architecture reorganization.
- To better review this PR, please do it commit-per-commit. The commit order was planned ahead for review easiness and some commits have explanations about the changes.
- A few files were moved to new folders. The commits that move files just include the movement diff to work well with git move detection.
- A lot of `.asmdef` files were changed. To refactor them faster, the `.asmdef` files were edited by hand. (seems that Rider still can't modify them through the refactor commands...).


## How to test it?
1. Go to: https://play.decentraland.zone/index.html?renderer=urn:decentraland:off-chain:renderer-artifacts:refactor/move-world-runtime-interfaces-to-own-assembly&ENV=org
2. The client should run correctly. If the world loads then most likely this PR is working correctly.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
